src/app/projects/components/project-settings/components/data-schema/data-schema.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { first } from 'rxjs/operators';
+import { Project } from 'src/app/base/entities/project';
+import { ProjectApolloService } from 'src/app/base/services/project/project-apollo.service';
+import { attributeVisibilityStates, getTooltipVisibilityState } from '../../../create-new-attribute/attributes-visibility-helper';
+import { Attribute, AttributeVisibilityStates } from '../../entities/attribute.type';
+
+
+@Component({
+  selector: 'kern-data-schema',
+  templateUrl: './data-schema.component.html',
+  styleUrls: ['./data-schema.component.scss']
+})
+export class DataSchemaComponent implements OnChanges {
+
+  @Input() project: Project;
+  @Input() attributeVisibilityStates: AttributeVisibilityStates;
+  @Input() attributes: Attribute[];
+  @Input() pKeyValid: boolean;
+
+  tooltipsArray: string[] = [];
+
+  constructor(private projectApolloService: ProjectApolloService) {
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    this.attributes = changes.attributes?.currentValue;
+    this.pKeyValid = changes.pKeyValid?.currentValue;
+    this.attributes.forEach((attribute) => {
+      this.tooltipsArray.push(getTooltipVisibilityState(attribute.visibility));
+    });
+  }
+
+  updateVisibility(option: string, attribute: Attribute) {
+    const visibility = attributeVisibilityStates[option].value;
+    this.projectApolloService.updateAttribute(this.project.id, attribute.id, attribute.dataType, attribute.isPrimaryKey, attribute.name, attribute.sourceCode, visibility).pipe(first()).subscribe();
+  }
```

Comment:
old code, no need to provide everything. project-id, attribute id and visibility would do it (since optional parameters in the mutation on backend) should be possible by setting the method variables to null for everything but the nessecary
```suggestion
    this.projectApolloService.updateAttribute(this.project.id, attribute.id, null, null, null, null, visibility).pipe(first()).subscribe();
  }
```